compiler/rustc_metadata/src/creader.rs

```diff
@@ -908,6 +908,10 @@ impl<'a> CrateLoader<'a> {
                 // Don't worry about pathless `--extern foo` sysroot references
                 continue;
             }
+            if entry.nounused_dep {
```

Comment:
hmm... would something like `ignore_unused` be a better name for this? I know the convention in other compilers is `X` and `noX` for enabling or disabling `X`, but it's kinda hard to understand whether `nounused` is denying unused or allowing unused.